Build: Fix the sb-bench CI step

ndelangen opened the pull request:
Issue: The CI task of sb-bench was failing.

- I had to update and sync up the dependencies of playwright with the ones in the image used for the CI runner.
- I had to fix some issues after the emotion upgrade because I had to regenerate the yarn lockfile to do the above.

related: #18992 I think this replaces it. @Andarist

ndelangen committed:
try to fix the sb-becnh ci step

ndelangen committed:
found the actual fix: newer version of npm is stricter about npx usage

shilman reviewed (approved):
You are the king!!! 👑

ndelangen committed:
upgrades

ndelangen committed:
fix the build system.

Emotion changed to use exports map, and esbuild wasn't pivking the right entry anymore,
resulting in react not being detected, and thus not bundled, and thus missing at runtime!

I changed the tsup/esbuild config to prefer the ESM version, and this seems to have fixed the issue.

Many thanks to @Anderist for the assist!

ndelangen committed:
'fix' warnings

ndelangen committed:
sync playwright version with pre-installed in docker and from npm

ndelangen committed:
so it seems that in production mode this is still an issue, so I did the suggested fix by @Anderist
He suggested that 'hiding' the dependency on an export that may or may not exists away from bundlers would solve it.
This is what they do in emotion itself, but esbuild optimizes that away again.

Andarist commented on scripts/prepare/bundle.ts:
```diff
@@ -80,6 +80,7 @@ const run = async ({ cwd, flags }: { cwd: string; flags: string[] }) => {
           : false,
       esbuildOptions: (c) => {
         /* eslint-disable no-param-reassign */
+        c.conditions = ['module'];
```
I would recheck what kind of an effect does it have on esbuild - is this purely additive to the default conditions? or does it replace them? I think that I was checking this in the past and that this is additive - some default conditions just cant be turned off but I'm not 100% sure.

Andarist commented:
> related: https://github.com/storybookjs/storybook/pull/18992 I think this replaces it. @Andarist

Yes, this is same-ish - I would still recommend updating the dependency ranges for Emotion deps in `package.json` files though.

ndelangen commented:
So it seems there's a bunch of these missing / broken:
https://www.chromatic.com/test?appId=5da9c7ef6e6df60020a7f102&id=6308f03bd7fcbd4674f94f90

ndelangen committed:
fix

ndelangen committed:
Merge branch 'next' into tech/fix-bench

# Conflicts:
#	code/yarn.lock

ndelangen committed:
bump sb-bench version

ndelangen committed:
upgrade sb-bench again, this might fix it!

ndelangen committed:
update

ndelangen commented:
@shilman @tmeasday good to merge?
